### DISCUSSION TITLE
How to fine-tune Faster Whisper?

### DISCUSSION THREAD

**Discussion by toanhuynhnguyen** [SYSTRAN/faster-whisper#1058] [Q&A]
Dear everyone,
How to fine-tune Faster Whisper for proper nouns and terminologies?
Thanks.

**Comment by Hikari-Tsai** [comment]
faster-whisper is an inference implementation; it does not provide a training loop. Fine-tune a Transformers-compatible Whisper model with PyTorch/Hugging Face, then convert the resulting checkpoint to CTranslate2.

For example, after fine-tuning and saving to `./whisper-domain-model`:

```bash
pip install 'transformers[torch]>=4.23' ctranslate2

ct2-transformers-converter \
  --model ./whisper-domain-model \
  --output_dir ./whisper-domain-model-ct2 \
  --copy_files tokenizer.json preprocessor_config.json \
  --quantization float16
```

Then load it normally:

```python
from faster_whisper import WhisperModel

model = WhisperModel("./whisper-domain-model-ct2", device="cuda", compute_type="float16")
```

For proper nouns, try `hotwords` or `initial_prompt` before fine-tuning; they are far cheaper and often sufficient:

```python
segments, info = model.transcribe(
    audio,
    hotwords="AcmeCloud ZetaSync Hikari Tsai",
)
```

Evaluate on a held-out domain test set. Fine-tuning on a small vocabulary list without representative audio can damage general recognition and does not guarantee that rare names will always be emitted.
